DatadogSessionReplay/Sources/SessionReplay+Internal.swift

```diff
@@ -0,0 +1,152 @@
+    case invisibleElement
+}
+
+@_spi(Internal) open class CustomNodeRecorder: NodeRecorder {
+    public let identifier = UUID()
+    
+    public init () {}
+
+    internal func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeRecordingContext) -> NodeSemantics? {
+        let element = getSessionReplayElement(view: view, attributes: attributes, context: context)
+        switch element {
+        case .invisibleElement:
+            return InvisibleElement.constant
+        case .none:
+            return nil
+        case .specificElement(let subtreeStrategy, let attributes, let builder):
+            let nodes = [
+                Node(viewAttributes: attributes, wireframesBuilder: builder)
+            ]
+            return SpecificElement(subtreeStrategy: subtreeStrategy, nodes: nodes)
+        }
+    }
+
+    // will be overriden with actual implementation
+    open func getSessionReplayElement(
+        view: UIView,
+        attributes: SessionReplayViewAttributes,
+        context: SessionReplayViewTreeRecordingContext
+    ) -> SessionReplayElement? {
+        return nil
+    }
+}
+
+@_spi(Internal) public struct GenericTextWireframesBuilder: NodeWireframesBuilder {
+    public let id: NodeIDGenerator
+    public let frame: CGRect
+    public let text: String
+    public let textAlignment: TextAlignment
+    public let clip: SessionReplayContentClip
+    public let textColor: CGColor?
+    public let font: UIFont
+    public let fontScalingEnabled: Bool
+    public let borderColor: CGColor?
+    public let borderWidth: CGFloat?
+    public let backgroundColor: CGColor?
+    public let cornerRadius: CGFloat?
+    public let opacity: CGFloat?
+    public let view: UIView
+    public let nodeRecorder: CustomNodeRecorder
+    
+    public let wireframeRect: CGRect
+    
+    public init(id: NodeIDGenerator, frame: CGRect, text: String, textAlignment: TextAlignment, clip: SessionReplayContentClip, textColor: CGColor?, font: UIFont, fontScalingEnabled: Bool, borderColor: CGColor?, borderWidth: CGFloat?, backgroundColor: CGColor?, cornerRadius: CGFloat?, opacity: CGFloat?, view: UIView, nodeRecorder: CustomNodeRecorder, wireframeRect: CGRect) {
+        self.id = id
+        self.frame = frame
+        self.text = text
+        self.textAlignment = textAlignment
+        self.clip = clip
+        self.textColor = textColor
+        self.font = font
+        self.fontScalingEnabled = fontScalingEnabled
+        self.borderColor = borderColor
+        self.borderWidth = borderWidth
+        self.backgroundColor = backgroundColor
+        self.cornerRadius = cornerRadius
+        self.opacity = opacity
+        self.view = view
+        self.nodeRecorder = nodeRecorder
+        self.wireframeRect = wireframeRect
+    }
+
+    internal func buildWireframes(with builder: WireframesBuilder) -> [SRWireframe] {
```

Comment:
We cannot expose `SRWireframe` since it is declared internal in the Data models.
The idea here is to create a `NodeWireframesBuilder` class that is initialized with all the needed information for the builder to be called.

Reply:
I start to see that the design here is a little bit awkward on the native side. It almost asks for making exact copies of model, which you actually ended up doing, but it's just "hidden" behind `GenericWireframeBuilder`.

IMO worth brainstorming some architecture changes to make it scale better.

Saying that - for a single wireframe builder I think it's fine.